core/src/main/java/org/bitcoinj/core/TransactionOutPoint.java

```diff
@@ -139,6 +189,36 @@ public byte[] bitcoinSerialize() {
         public int getMessageSize() {
             return BYTES;
         }
+    }
+
+    class Connected extends Unconnected implements TransactionOutPoint  {
+
+        /** Special outpoint that normally marks a coinbase input. It's also used as a test dummy. */
+        public static final Connected UNCONNECTED =
+                new Connected(ByteUtils.MAX_UNSIGNED_INTEGER, Sha256Hash.ZERO_HASH);
```

Comment:
Why is this in `Connected` rather than in `Unconnected`?

Reply:
This is temporary. At present `Connected` class can be either "connected" or "unconnected" based on mutable fields in the class. There is more conversion that needs to be done in the other classes before it can fully use the new semantics.

Reply:
The mutability would be taken care of via #2997 – we can merge that either before or after your PR, I'm happy to rebase if needed.

Reply:
But going back to my above concern: I think even in this temporary state, the `UNCONNECTED` instance should go either into the `Unconnected` subclass or simply stay at the `TransactionOutPoint` interface level.